### PR TITLE
Template 4.2 updates + troubleshooting

### PR DESCRIPTION
**Pull request by danopolan**
Updates to match ISTQB Template v4.2
Troubleshooting section added

### CODE REVIEW

File: example-document/03-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+```
+yaml.scanner.ScannerError: mapping values are not allowed in this context
+  in "/__w/istqb-ctal-ta/istqb-ctal-ta/sample-exam-answers/questions.yml", line 562, column 19
+```
+To resolve these issues:
+1. Navigate to affected file (`sample-exam-answers/questions.yml` line 562, position 19 in this case)
+2. Resolve the formatting issue which could be cause by several things. The most common are:
+  * wrong indents - fix by adding 2 spaces per indent level
+  * using colon `:` or hashtag `#` in the text - fix by adding the text in quotes `"` or `'`
+  * missing pipe `|` in right after the key name to allow multiline inputs - fix by adding pipe `|` before multiline markdown input
```
Comment (Witiko) [inline example-document/03-troubleshooting.md:26]: This section seems out-of-date after https://github.com/istqborg/istqb_product_base/pull/87.

File: example-document/03-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+6. Seach for commonly known error messages using native GitHub log search on top right corner
+
+## Validate YAML documents
+
+YAML formatting is strict, and all errors in this job are related to it. Search for `yaml.scanner.ScannerError` to see what went wrong. Example of the log:
+
+```
+yaml.scanner.ScannerError: mapping values are not allowed in this context
+  in "/__w/istqb-ctal-ta/istqb-ctal-ta/sample-exam-answers/questions.yml", line 562, column 19
+```
```
Comment (Witiko) [inline example-document/03-troubleshooting.md:20]: This code example overflows:

image

In general, code examples should not exceed 90 characters per line.

Here, I suggest shrinking the path:

````suggestion
```
yaml.scanner.ScannerError: mapping values are not allowed in this context
  in "istqb-ctal-ta/sample-exam-answers/questions.yml", line 562, column 19
```
````

File: example-document/03-troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+```
+Latexmk: ====Undefined refs and citations with line #s in .tex file:
+  Reference `section:analyze-sut-to-determine-the-appropriate-tas' on page 21 undefined on input line 37
+  Reference `section:apply-layering-of-taf' on page 22 undefined on input line 38
+  Reference `section:apply-design-principles-and-design-patterns-in-ta' on page 24 undefined on input line 38
+  Reference `section:apply-different-approaches-for-automating-tcs' on page 47 undefined on input line 43
+  Reference `section:apply-layering-of-taf' on page 48 undefined on input line 43
+  Reference `section:explain-which-factors-support-and-affect-tas-maintainability' on page 50 undefined on input line 43
+```
```
Comment (Witiko) [inline example-document/03-troubleshooting.md:42]: This code example overflows:

image

In general, code examples should not exceed 90 characters per line.

Here, I suggest line-breaking the logs:

````suggestion
```
Latexmk: ====Undefined refs and citations with line #s in .tex file:
  Reference `section:analyze-sut-to-determine-the-appropriate-tas'
    on page 21 undefined on input line 37
  Reference `section:apply-layering-of-taf'
    on page 22 undefined on input line 38
  Reference `section:apply-design-principles-and-design-patterns-in-ta'
    on page 24 undefined on input line 38
  Reference `section:apply-different-approaches-for-automating-tcs'
    on page 47 undefined on input line 43
  Reference `section:apply-layering-of-taf'
    on page 48 undefined on input line 43
  Reference `section:explain-which-factors-support-and-affect-tas-maintainability'
    on page 50 undefined on input line 43
```
````

File: example-document/03-troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+## Invalid references to files in `syllabus.tex` file
+
+Every MD files you want to include in your document have to be added to TEX file. If the file referenced in the TEX file is not existent (wrong path, wrong name of the file, missing extension, etc.) the pipeline job will fail. Example of the log:
+
+```
+! Package markdown Error: Markdown file
+(markdown)               
+syllabus/01-tasks-in-**test**-process.md
+(markdown)                does not exist
+```
```
Comment (Witiko) [inline example-document/03-troubleshooting.md:60]:
````suggestion
```
! Package markdown Error: Markdown file
(markdown)                syllabus/01-tasks-in-**test**-process.md
(markdown)                does not exist
```
````

File: example-document/03-troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+These errors can also seem in *Produce DOCX documents* pipeline job. But the output here will show only the name of the first file missing. Example of the log:
+
+```
+FileNotFoundError: [Errno 2] No such file or directory: '/__w/istqb-ctal-ta/istqb-ctal-ta/syllabus/01-tasks-in-**test**-process.md'
+```
```
Comment (Witiko) [inline example-document/03-troubleshooting.md:66]: This code example overflows:

image

In general, code examples should not exceed 90 characters per line.

Here, I suggest line-breaking the logs:

````suggestion
```
FileNotFoundError: [Errno 2] No such file or directory:
  '/__w/istqb-ctal-ta/istqb-ctal-ta/syllabus/01-tasks-in-**test**-process.md'
```
````

**Review by Witiko — CHANGES_REQUESTED** [review]

File: example-document/03-troubleshooting.md (hunk 1 above)
Comment (danopolan) [reply example-document/03-troubleshooting.md]: Section rewritten.

File: example-document/03-troubleshooting.md (hunk 3 above)
Comment (danopolan) [reply example-document/03-troubleshooting.md]: Fixed

File: example-document/03-troubleshooting.md (hunk 4 above)
Comment (danopolan) [reply example-document/03-troubleshooting.md]: Fixed

File: example-document/03-troubleshooting.md (hunk 5 above)
Comment (danopolan) [reply example-document/03-troubleshooting.md]: Fixed

File: example-document/03-troubleshooting.md (hunk 2 above)
Comment (danopolan) [reply example-document/03-troubleshooting.md]: Fixed

File: example-document/03-troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	    on page 22 undefined on input line 38
+	  Reference `section:apply-design-principles-and-design-patterns-in-ta'
+	    on page 24 undefined on input line 38
+	  Reference `section:apply-different-approaches-for-automating-tcs'
+	    on page 47 undefined on input line 43
+	  Reference `section:apply-layering-of-taf'
+	    on page 48 undefined on input line 43
+	  Reference `section:explain-which-factors-support-and-affect-tas-maintainability'
+	    on page 50 undefined on input line 43
+	```
```
Comment (Witiko) [inline example-document/03-troubleshooting.md:48]: This does not properly close the code block:

image

To fix this, you should remove the indent to align the closing ticks with the opening ticks.

**Review by Witiko — CHANGES_REQUESTED** [review]

**Comment by Witiko** [comment]
@danopolan I noticed a factual error in the last troubleshooting section. The factual error is that while the log shows the filename `01-tasks-in-test-process.md` whereas the TeX file shows the filename `01-tasks-in-the-test-process.md`, which is inconsistent. Furthermore, I noticed a use of `**` in the fenced block, which will not be rendered in Markdown.

I fixed the explanation, so that the log and the TeX file show the same filename but the MD file is named differently on-disk. Furthermore, I used natural language what was previously only implied by `**`. Lastly, I replaced all occurences of TEX by the TeX logotype:

image

Do let me know if you are OK with the changes.

**Pull request merged**